From 20fd1ae415683168b2dc5ad707ea422c67429578 Mon Sep 17 00:00:00 2001
From: Pooja Prajod <a0132412@ti.com>
Date: Fri, 12 May 2017 16:12:55 +0530
Subject: [gst-bad] gstwaylandsink: Fix scale up with padded video

Usecases which expects waylandsink cropping support
negotiates with padded height and width. After cropping,
the destination viewport is set with the negotiated
height and width which results in scaled up output.
To fix this the args while setting dest viewport should
be taken as the height and width present in the crop metadata

Signed-off-by: Pooja Prajod <a0132412@ti.com>
---
 ext/wayland/wlwindow.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/ext/wayland/wlwindow.c b/ext/wayland/wlwindow.c
index f7e3324..decfdf8 100644
--- a/ext/wayland/wlwindow.c
+++ b/ext/wayland/wlwindow.c
@@ -278,7 +278,13 @@ gst_wl_window_resize_video_surface (GstWlWindow * window, gboolean commit)
   gst_video_sink_center_rect (src, dst, &res, TRUE);
 
   wl_subsurface_set_position (window->video_subsurface, res.x, res.y);
-  wl_viewport_set_destination (window->video_viewport, res.w, res.h);
+
+  if (window->display->crop) {
+    wl_viewport_set_destination (window->area_viewport, window->display->crop->width,  window->display->crop->height);
+    wl_viewport_set_destination (window->video_viewport, window->display->crop->width,  window->display->crop->height);
+  } else {
+    wl_viewport_set_destination (window->video_viewport, res.w, res.h);
+  }
 
   if (commit) {
     wl_surface_damage (window->video_surface, 0, 0, res.w, res.h);
-- 
1.9.1

